### PR TITLE
test: Validate nerserver availability before testing connectivity

### LINKED ISSUE #7673: CI: runtime.RuntimeConnectivityInVethModeTest Basic Connectivity test Test connectivity between containers without policies imported
## CI: runtime.RuntimeConnectivityInVethModeTest Basic Connectivity test Test connectivity between containers without policies imported

Observed in PR #7671, but given the changes seems that it may be unrelated to the PR changes and may affect master.

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11480/testReport/junit/(root)/runtime/RuntimeConnectivityInVethModeTest_Basic_Connectivity_test_Test_connectivity_between_containers_without_policies_imported/

#### Stacktrace
```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:410
Expected command: docker exec -i  client netperf -c -C -t TCP_SENDFILE -H f00d::a0f:0:0:3621 
To succeed, but it failed:
Exitcode: 1 
Stdout:
 	 TCP SENDFILE TEST from ::0 (::) port 0 AF_INET6 to f00d::a0f:0:0:3621 () port 0 AF_INET6
	 
Stderr:
 	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/runtime/connectivity.go:130
```

#### Standard Output
```
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 0
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Number of "Goroutine took lock for more than" in logs: 0
No errors/warnings found in logs

```

#### Standard Error
```
cleaning up VMs started for runtime tests
=== Test Finished at 2019-04-11T02:49:33Z====
===================== TEST FAILED =====================
cmd: sudo cilium endpoint list
Exitcode: 1 
Stdout:
 	 
Stderr:
 	 

===================== Exiting AfterFailed =====================
STEP: removing container client
STEP: removing container server

[[ATTACHMENT|4395e57e_RuntimeConnectivityInVethModeTest_Basic_Connectivity_test_Test_connectivity_between_containers_without_policies_imported.zip]]
```

[4395e57e_RuntimeConnectivityInVethModeTest_Basic_Connectivity_test_Test_connectivity_between_containers_without_policies_imported.zip](https://github.com/cilium/cilium/files/3066793/4395e57e_RuntimeConnectivityInVethModeTest_Basic_Connectivity_test_Test_connectivity_between_containers_without_policies_imported.zip)

### PR DESCRIPTION
**Pull request by tgraf**
Fixes: #7673

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7905)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by ianvernon** [comment]
Boot VMs stage timed out again: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/12021/

**Comment by ianvernon** [comment]
test-me-please

File: test/runtime/connectivity.go [hunk 1]
```diff
@@ -93,6 +93,15 @@ var runtimeConnectivityTest = func(datapathMode string) func() {
...
 				Expect(err).Should(BeNil())
+
+				err = helpers.WithTimeout(func() bool {
+					res := vm.ContainerExec(helpers.Server, "netperf -H 127.0.0.1 -l 1")
+					if !res.WasSuccessful() {
+						logger.Info("Waiting for netserver to come up")
+					}
+					return res.WasSuccessful()
+				}, "netserver did not come up in time", &helpers.TimeoutConfig{Timeout: 20 * time.Second})
+				Expect(err).Should(BeNil())
```
Comment (eloycoto) [inline test/runtime/connectivity.go]: I would add here an assert message, will provide context in the Junit output.

Comment (tgraf) [reply test/runtime/connectivity.go]: Done. It was already pretty good to debug when I tested it:
```
RuntimeConnectivityInVethModeTest
/Users/tgraf/dev/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:378
  Basic Connectivity test
  /Users/tgraf/dev/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:378
    Test NAT46 connectivity between containers [BeforeEach]
    /Users/tgraf/dev/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:410

    Expected
        <*errors.errorString | 0xc00033abe0>: {
            s: "Timeout reached: netserver did not come up in time",
        }
    to be nil

    /Users/tgraf/dev/go/src/github.com/cilium/cilium/test/runtime/connectivity.go:104
```

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
VM image download timed out

**Comment by tgraf** [comment]
test-me-please

File: test/runtime/connectivity.go [hunk 2]
```diff
@@ -93,6 +93,15 @@ var runtimeConnectivityTest = func(datapathMode string) func() {
 					return true
 				}, "Endpoints are not ready", &helpers.TimeoutConfig{Timeout: 150 * time.Second})
 				Expect(err).Should(BeNil())
+
+				err = helpers.WithTimeout(func() bool {
```
Comment (aanm) [inline test/runtime/connectivity.go:97]: `WaitForNPods` checks if the container is ready. Shouldn't that be enough?

**Review by aanm — APPROVED** [review]

File: test/runtime/connectivity.go (hunk 2 above)
Comment (tgraf) [reply test/runtime/connectivity.go:97]: These are runtime tests, there is no k8s that provides a readiness probe

**Pull request merged**